deploy/kubernetes/examples/demo-storageclass-using-secret-from-configuration.yaml

```diff
@@ -0,0 +1,25 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: demo-storageclass
+provisioner: block.csi.ibm.com
+volumeBindingMode: WaitForFirstConsumer  # Optional. Required for Topology Awareness support.
+parameters:
+  # a pool value MUST be provided for all systems.
+  # pool, SpaceEfficiency and volume_name_prefix can be provided in by_system_id parameter for each system or by parameter name as default.
+
+  by_system_id: '{"demo-system-id1":{"pool":"demo-pool","SpaceEfficiency":"thin","volume_name_prefix":"demoPVC"},
+                  "demo-system-id2":{"pool":"demo-pool2","volume_name_prefix":"demoPVC2"}}'  # Optional.
+  pool: demo-pool                # Optional.
+  SpaceEfficiency: deduplicated  # Optional.
+
+  csi.storage.k8s.io/provisioner-secret-name: demo-secret-with-config
+  csi.storage.k8s.io/provisioner-secret-namespace: default
+  csi.storage.k8s.io/controller-publish-secret-name: demo-secret-with-config
+  csi.storage.k8s.io/controller-publish-secret-namespace: default
+  csi.storage.k8s.io/controller-expand-secret-name: demo-secret-with-config
+  csi.storage.k8s.io/controller-expand-secret-namespace: default

```

Comment:
could we follow the CSI-2885 approach for new class yamls, even before the rest of them are aligned?

Reply:
done, you want me to align it?

Reply:
if @zingero is ok with it, sure

Reply:
yes please

Reply:
done